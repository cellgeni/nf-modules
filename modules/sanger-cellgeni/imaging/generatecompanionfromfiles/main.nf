process IMAGING_GENERATECOMPANIONFROMFILES {
    tag "${meta.id}"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container
        ? 'quay.io/cellgeni/imagetileprocessor:0.1.16'
        : 'quay.io/cellgeni/imagetileprocessor:0.1.16'}"

    input:
    tuple val(meta), path(images, stageAs: "images/*"), val(file_regex)

    output:
    tuple val(meta), path("${out_companion_name}"), emit: companion
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    out_companion_name = "${prefix}.companion.ome"
    """
    generate_companion.py run \\
        -input_folder images \\
        -regex '${file_regex}' \\
        -out_companion_xml "${out_companion_name}" \\
        ${args} \\

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        imaging: \$(generate_companion.py version)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    out_companion_name = "${prefix}.companion.ome"
    """
    echo ${args}
    
    touch ${out_companion_name}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        imaging: \$(generate_companion.py version)
    END_VERSIONS
    """
}
