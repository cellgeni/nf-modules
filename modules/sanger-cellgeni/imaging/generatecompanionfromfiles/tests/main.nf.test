// nf-core modules test imaging/generatecompanionfromfiles
nextflow_process {

    name "Test Process IMAGING_GENERATECOMPANIONFROMFILES"
    script "../main.nf"
    process "IMAGING_GENERATECOMPANIONFROMFILES"

    tag "modules"
    tag "modules_sangercellgeni"
    tag "imaging"
    tag "imaging/generatecompanionfromfiles"

    test("PE images - ome.tif") {

        when {
            params {
                module_args = ""
            }
            process {
                """
                input[0] = [
                    [ id:'PE' ], // meta map
                    file(params.modules_testdata_base_path + 'PerkinElmer/for_companion_generation/*tif', checkIfExists: true),
                    "*.ome.tif"
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("PE images - ome.tif - stub") {

        options "-stub"

        when {
            params {
                module_args = ""
            }
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + 'PerkinElmer/deconvolved/Well:0:0/*tif', checkIfExists: true),
                    "*.ome.tif",
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
