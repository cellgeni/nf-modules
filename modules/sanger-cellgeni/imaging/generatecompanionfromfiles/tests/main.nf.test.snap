{
    "PE images - ome.tif - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.companion.ome:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,2c13973c13bfc633ac0a7917e5d32d71"
                ],
                "companion": [
                    [
                        {
                            "id": "test"
                        },
                        "test.companion.ome:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2c13973c13bfc633ac0a7917e5d32d71"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-08-07T16:28:26.374059054"
    },
    "PE images - ome.tif": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "companion": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-08-07T15:19:38.339326222"
    }
}